### ISSUE TITLE
1.4.0: Exploded query params should not be auto-generated as a object query param

### ISSUE THREAD

**Issue by StephenOTT** [micronaut-projects/micronaut-openapi#155]
microanut 1.3.0
kotlin 1.3.61
micronaut open api 1.4.0

consider the following

```kotlin
    @Get(value = "/collections/{collectionId}/objects{?qParams*}")
    @Operation(
            summary = "Get all objects from a collection",
            description = "This Endpoint retrieves objects from a Collection. Clients can search for objects in the Collection, retrieve all objects in a Collection, or paginate through objects in the Collection.",
            security = [
                SecurityRequirement(name = "basicAuth")
            ]
    )
    @Parameters(
            Parameter(name = "apiRoot", `in` = ParameterIn.PATH, required = true, description = "the base URL of the API Root", schema = Schema(type = "string")),
            Parameter(name = "collectionId", `in` = ParameterIn.PATH, required = true, description = "the \u200Bidentifier\u200B of the Collection being requested", schema = Schema(type = "string")),
            Parameter(name = "added_after", `in` = ParameterIn.QUERY, description = "a single timestamp", example = "..."),
            Parameter(name = "limit", `in` = ParameterIn.QUERY, description = "a single integer", example = "..."),
            Parameter(name = "next", `in` = ParameterIn.QUERY, description = "a single string", example = "..."),
            Parameter(name = "match[id]", `in` = ParameterIn.QUERY, description = "an id(s) of an object", example = "..."),
            Parameter(name = "match[type]", `in` = ParameterIn.QUERY, description = "the type(s) of an object", example = "..."),
            Parameter(name = "match[version]", `in` = ParameterIn.QUERY, description = "the version(s) of an object", example = "..."),
            Parameter(name = "match[spec_version]", `in` = ParameterIn.QUERY, description = "the specification version(s)", example = "..."),
            Parameter(name = Headers.ACCEPT, `in` = ParameterIn.HEADER, required = false, schema = Schema(type = "string", allowableValues = [TaxiiMediaType.APPLCATION_JSON_TAXII_VERSION_2_1, TaxiiMediaType.APPLCATION_JSON_TAXII_VERSION_2_0, TaxiiMediaType.APPLCATION_JSON_TAXII], defaultValue = TaxiiMediaType.APPLCATION_JSON_TAXII_VERSION_2_1))
    )
    open fun getCollectionObjects(apiRoot: String, collectionId: String, @QueryValue qParams: Map<String, Any>?): Single<HttpResponse<Envelop>> {
   ...
}
```

The exploded query params as part of the 1.4.0 started to get generated as 

```yml
      - name: qParams
        in: query
        required: false
        schema:
          type: object
          additionalProperties: true
          nullable: true
```

Should there is a additional annotation to indicate parameters to ignore in the micronaut auto get of the openapi yml?  As the Parameters config in the Operation annotation has all of the specific details..

**Comment by croudet** [comment]
Have you tried to annotate it with `@Hidden`?

**Comment by StephenOTT** [comment]
Hidden was not required: hidden can't actually be used on a param. 

The issue is with definition of parameters within the Operation annotation vs using the Parameters annotation at the fiction level. 

If you use the Parameters annotation at the top level, the queryValue is shown. But if you use the Operation annotation and define the parameters inside of that annotation, then it works as expected. 

This seems like a bug with the parsing of the Parameters annotation at the fuction level

**Comment by ilopmar** [comment]
This is the same as in #416 and it's fixed by #545.

Please keep in mind that you need to annotate `qParam` with `@Parameter(hidden = true)` to skip it.
